packages/desktop/src/providers/network/provider.rs

```diff
@@ -2,44 +2,58 @@ use std::sync::Arc;
 
 use anyhow::Result;
 use async_trait::async_trait;
-use sysinfo::Networks;
+use netdev::interface::{
+  get_default_interface, get_interfaces, Interface,
+};
 use tokio::{sync::Mutex, task::AbortHandle};
 
 use crate::providers::{
   interval_provider::IntervalProvider,
-  network::{NetworkInterface, NetworkVariables},
+  network::parse_netsh::get_primary_interface_ssid_and_strength,
   variables::ProviderVariables,
 };
 
-use super::NetworkProviderConfig;
+use super::{
+  Gateway, NetworkInterface, NetworkProviderConfig, NetworkVariables,
+};
 
 pub struct NetworkProvider {
   pub config: Arc<NetworkProviderConfig>,
   abort_handle: Option<AbortHandle>,
-  networks: Arc<Mutex<Networks>>,
+  netdev_data: Arc<Mutex<NetdevData>>,
+}
+
+pub struct NetdevData {
+  pub default_interface: Arc<Mutex<Interface>>,
+  pub interfaces: Arc<Mutex<Vec<Interface>>>,
 }
 
 impl NetworkProvider {
   pub fn new(config: NetworkProviderConfig) -> NetworkProvider {
     NetworkProvider {
       config: Arc::new(config),
       abort_handle: None,
-      networks: Arc::new(Mutex::new(Networks::new_with_refreshed_list())),
+      netdev_data: Arc::new(Mutex::new(NetdevData {
+        default_interface: Arc::new(Mutex::new(
+          get_default_interface().unwrap(),
+        )),
+        interfaces: Arc::new(Mutex::new(get_interfaces())),
+      })),
```

Comment:
we shouldn't get the default interface + interfaces in `new`, since immediately afterwards `get_refreshed_variables` will be called on the provider. if there's no state that needs to be shared in between every refresh, you can do:

```rust
pub struct NetworkProvider {
  pub config: Arc<NetworkProviderConfig>,
  abort_handle: Option<AbortHandle>,
  state: Arc<Mutex<()>>,
}
```

Reply:
I struggled with this element as the netdev crate doesn't provide a "manager" that you can then get the correct variables from - passing that in as the "state" variable is a pattern followed by the other providers e.g., with sysinfo

so instead I passed the actual variables through as it's own struct and initialized there

can you describe what change you mean to make as I am struggling to get it to work with your suggested change to the NetworkProvider

Reply:
you can remove the `netdev_data` field from the struct and then modify:

```rust
impl IntervalProvider for NetworkProvider {
  type State = ();

  fn state(&self) -> Arc<()> {
    Arc::new(())
  }

  async fn get_refreshed_variables(
    _: &NetworkProviderConfig,
    __: &(),
  ) -> Result<ProviderVariables> {
    ...
  }
}
```

Reply:
cheers that makes a lot more sense and is a lot easier! can take that into the next component I'm building too...